### ISSUE TITLE
enhancement to have keyboard command to save checkpoint now

### ISSUE THREAD

**Issue by wrapperband** [jcjohnson/torch-rnn#152]
The interface has been changed from char-rnn, so I didn't set the save checkpoint correctly. I've got a larger nn size started with tensor-rnn and it looks like it won't do a checkpoint till 50 epochs.

It's currently on 36 and is calculating 2 epochs per day (250 buffer), ie another week.

Looking at the learning rate it would have been great to save a checkpoint "on demand" i.e. via keyboard command at certain points, 18 epochs, being the point where the training leveled of, in this case.

**Comment by dgcrouse** [comment]
Trying to pull interactive keyboard commands is a bit out of scope for this project. Having a loop checking for keyboard commands then passing that on at the end of a batch will increase complexity significantly and will only be useful in a few edge cases. If you would like, you can submit a pull request, but I just don't think that will be on the roadmap anytime soon for us to implement.
